### DISCUSSION TITLE
Ruby bundler update upgrades gem dependency to a low version

### DISCUSSION THREAD

**Discussion by DawidJanczak** [renovatebot/renovate#24370] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

Hello,

We ran into an interesting problem the other day. This version of sequel gem https://rubygems.org/gems/sequel/versions/5.72.0 adds `bigdecimal` library with a `>= 0` constraint. Now because of how renovate runs the upgrade: `bundler lock --minor --strict --update sequel` and because of the way `bundler lock --minor` works the `bigdecimal` library got set to a very old version 1.4.4 instead of the newest 3.1.4 and that old version fails to load on newer Ruby. I _think_ it's the way `bundler lock --minor --strict` works and that itself seems a bit odd (I'll file a separate ticket there), but I was wondering if renovate has to pass the `--minor` and `--strict` flags in case of this upgrade.

If I understand the logic here renovate first checks if upgrade should be major, minor or patch and then upgrade gems separately depending on that. So in our case we have a `minor` update to `sequel`, but also `patch` updates to some other libs (`bundler lock --patch --strict --update activesupport pg rubocop && bundler lock --minor --strict --update sequel`)

I'm wondering if this is really necessary or could we just run `bundle lock --update`? Wouldn't that end up with the same logic _for the gems involved_? The dependencies would then get upgraded to latest possible as well, which is what we usually want.

Let me know if this is unclear on its own and I'll try to prepare a reproduction repository with this later today or tomorrow.

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: 12 flattened updates found: activesupport, pg, sequel, rubocop, gcr.io/kaniko-project/executor, gcr.io/kaniko-project/executor, actions/checkout, actions/checkout, haproxy, postcss
.
.
.
          {
            "changelogUrl": "https://sequel.jeremyevans.net/rdoc/files/CHANGELOG.html",
            "currentVersion": "5.71.0",
            "datasource": "rubygems",
            "depName": "sequel",
            "fixedVersion": "5.71.0",
            "homepage": "https://sequel.jeremyevans.net",
            "isSingleVersion": true,
            "lockedVersion": "5.71.0",
            "managerData": {
              "lineNumber": 14
            },
            "packageName": "sequel",
            "registryUrl": "https://rubygems.org",
            "sourceUrl": "https://github.com/jeremyevans/sequel",
            "versioning": "ruby",
            "warnings": [],
            "updates": [
              {
                "bucket": "latest",
                "newVersion": "5.72.0",
                "releaseTimestamp": "2023-09-01T07:23:20.073Z",
                "newMajor": 5,
                "newMinor": 72,
                "updateType": "minor",
                "isRange": true,
                "isLockfileUpdate": true,
                "branchName": "renovate/all"
              }
            ]
          },
.
.
.
DEBUG: bundler.updateLockedDependency: sequel@5.71.0 -> 5.72.0 [Gemfile.lock] (branch="renovate/all")
.
.
.
DEBUG: Executing command (branch="renovate/all")
{
  "command": "docker run --rm --name=renovate_a_sidecar --label=renovate_a_child --memory=3584m -v \"/tmp/worker/66e0a1/b3a5a2/repos/github/ygt/sb-voucher-redemptions\":\"/tmp/worker/66e0a1/b3a5a2/repos/github/ygt/sb-voucher-redemptions\" -v \"/tmp/worker/66e0a1/b3a5a2/cache\":\"/tmp/worker/66e0a1/b3a5a2/cache\" -e GEM_HOME -e CONTAINERBASE_CACHE_DIR -w \"/tmp/worker/66e0a1/b3a5a2/repos/github/ygt/sb-voucher-redemptions\" ghcr.io/containerbase/sidecar:9.19.4 bash -l -c \"install-tool ruby 3.2.2 && install-tool bundler 2.3.26 && ruby --version && bundler lock --patch --strict --update activesupport pg rubocop && bundler lock --minor --strict --update sequel\""
}
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment, marked as answer]
The use of flags is intentional, to avoid the scenario where Renovate tells you it's giving you a patch update but bundler makes a minor change

**Reply by DawidJanczak** [reply]
Ah yes of course, thanks for clarifying.

**Reply by rarkins** [reply]
Do you have a reason that you _must_ use range constraints in your Gemfile and not fixed/pinned versions? Using pinned versions in the Gemfile makes these types of problems go away. Unless you are publishing a library to Rubygems where using range constraints is desirable, you'll get much less headaches with pinned versions.

**Reply by DawidJanczak** [reply]
I'm not sure that would help in this case? It would still be a `minor` upgrade wouldn't it?

**Reply by rarkins** [reply]
It would help
